File: src/lib/shared/classes/SpotifyUser.ts
import {ExplicitContent} from "./ExplicitContent";
import {ExternalUrls} from "./ExternalUrls";
import {Followers} from "./Followers";
import {Image} from "./Image";

export class SpotifyUser {
  country: string | null = null
  display_name: string | null = null
  email: string | null = null
  explicit_content: ExplicitContent | null = new ExplicitContent()
  external_urls: ExternalUrls | null = new ExternalUrls()
  followers: Followers | null = new Followers()
  href: string | null = null
  id: string | null = null
  images: Image[] = []
  product: string | null = null
  type: string | null = null
  uri: string | null = null

  constructor(obj: any) {

    if (obj === null) throw new Error("Object is null")

    this.country = obj?.country
    this.display_name = obj?.display_name
    this.email = obj?.email
    this.explicit_content = ExplicitContent.from(obj?.explicit_content)
    this.external_urls = ExternalUrls.from(obj?.external_urls)
    this.followers = Followers.from(obj?.followers)
    this.href = obj?.href
    this.id = obj?.id
    this.images = obj?.images?.map((value: any) => Image.from(value))
    this.product = obj?.product
    this.type = obj?.type
    this.uri = obj?.uri
  }
}

File: src/main/java/de/wittenbude/expandify/models/spotifydata/SpotifyUser.java
package de.wittenbude.expandify.models.spotifydata;

import com.neovisionaries.i18n.CountryCode;
import de.wittenbude.expandify.models.spotifydata.helper.SpotifyFollowers;
import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;
import se.michaelthelin.spotify.enums.ModelObjectType;
import se.michaelthelin.spotify.enums.ProductType;
import se.michaelthelin.spotify.model_objects.specification.Image;
import se.michaelthelin.spotify.model_objects.specification.User;

import java.util.Map;

@Data
@NoArgsConstructor
@Document(collection = "users")
public class SpotifyUser {
    private String birthdate;
    private CountryCode country;
    private String displayName;
    private String email;
    @Field
    private Map<String, String> externalUrls;
    private SpotifyFollowers followers;
    private String href;
    @Id
    private String id;
    private Image[] images;
    private ProductType product;
    private ModelObjectType type;
    private String uri;

    public SpotifyUser(User user) {
        this.birthdate = user.getBirthdate();
        this.country = user.getCountry();
        this.displayName = user.getDisplayName();
        this.email = user.getEmail();
        this.externalUrls = user.getExternalUrls().getExternalUrls();
        this.followers = new SpotifyFollowers(user.getFollowers());
        this.href = user.getHref();
        this.id = user.getId();
        this.images = user.getImages();
        this.product = user.getProduct();
        this.type = user.getType();
        this.uri = user.getUri();
    }
}
